### PR TITLE
Add UI labels for authenticator_plugin serializers

### PR DESCRIPTION
**Pull request by john-westcott-iv**
Introduce a MixIn class which will store the `ui_field_label` from the plugins' serializer fields.
Show the `ui_field_label` on the authenticator_plugin screen and default the value to `Undefined` if it was not properly added.
This field is for consumption by the UI to display the field names when configuring an authenticator_plugin.

### CODE REVIEW

File: ansible_base/authenticator_plugins/base.py [hunk 1]
```diff
@@ -16,6 +17,7 @@ class BaseAuthenticatorConfiguration(serializers.Serializer):
         help_text="Any additional fields that this authenticator can take, they are not validated and passed directly back to the authenticator",
         required=False,
         allow_null=True,
+        ui_field_label='Additional Authenticator Fields',
```
Comment (relrod) [inline ansible_base/authenticator_plugins/base.py]: Can/should we start _()ing new user-facing strings? We already import `_` above.

File: ansible_base/authenticator_plugins/keycloak.py [hunk 2]
```diff
@@ -18,15 +17,29 @@ class KeycloakConfiguration(BaseAuthenticatorConfiguration):
         help_text=_("Location where this app can fetch the user's token from."),
         default="https://keycloak.example.com/auth/realms/<my_realm>/protocol/openid-connect/token",
         allow_null=False,
+        ui_field_label='Keycloak Access Token URL',
```
Comment (relrod) [inline ansible_base/authenticator_plugins/keycloak.py]: And these. We already do it for `help_text` :sweat_smile:

File: ansible_base/authenticator_plugins/base.py [hunk 3]
```diff
@@ -29,7 +31,16 @@ def get_configuration_schema(self):
...
 
-            schema.append({"name": f, "help_text": field.help_text, "required": not field.allow_null, "default": default, "type": field.__class__.__name__})
+            schema.append(
+                {
+                    "name": f,
+                    "help_text": field.help_text,
+                    "required": not field.allow_null,
+                    "default": default,
+                    "type": field.__class__.__name__,
+                    "ui_field_label": getattr(field, 'ui_field_label', 'Undefined'),
```
Comment (relrod) [inline ansible_base/authenticator_plugins/base.py]: And here

```suggestion
                    "ui_field_label": getattr(field, 'ui_field_label', _('Undefined')),
```

**Review by relrod — APPROVED** [review]
This looks good to me. You can merge as-is if you like. But as mentioned inline, I'm wondering if it's a good time to start `_()`ing new strings.

We can go back and get the existing ones later on, but maybe new ones should start doing it so we're moving in the right direction.

**Comment by john-westcott-iv** [comment]
> But as mentioned inline, I'm wondering if it's a good time to start `_()`ing new strings.

Yea, we should probablly start doing that.

**Pull request merged**